integration-tests/src/server.js

```diff
@@ -20,26 +21,44 @@ const app = express()
 const server = http.createServer(app)
 server.setTimeout(serverTimeout)
 
+const firestore = new Firestore()
+
 const testTimeout = async timeout => {
   await delay(timeout)
   throw new Error(`Test timeout after ${timeout}ms`)
 }
 
+
+
 const runTest = fn => withPuppeteer(async ({ browser, context, name, page, req, timeout }) => {
   const targetEnvParams = { ...envs[req.query.targetEnv] }
+  const testGroup = req.query.testGroup
+  const startTime = new Date()
+  const host = req.headers.host
+  const reportTest = async error => {
+    const endTime = new Date()
+    const testReport = { testGroup, runtimeInMilliseconds: endTime - startTime, status: error ? 'error' : 'success', testName: name, error, host, startTime }
+    await firestore.collection('tests').doc().create(testReport)
+  }
+  if (host === 'terra-bueller.appspot.com') {
+    throw new Error('Do not call without a version specified')
```

Comment:
I don't want to do that, because it could change mid-script. (If someone is deploying to the terra-bueller with --promote while a perf-test run script is in progress)

Reply:
To clarify, I was talking about being able to lock down the Terra UI build. Right now, we can only specify an env such as `dev` but we might want to allow pointing at a specific GAE version such as `https://20200619t224522-dot-bvdp-saturn-dev.appspot.com/`.